### DISCUSSION TITLE
[GraphQL API] How to create a branch protection that allows force pushes?

### DISCUSSION THREAD

**Discussion by douglascayers** [community/community#24728] [Apps, API and Webhooks] [answered]
<p>I’m migrating from the REST API v3 to the GraphQL API v4.</p>
<p>Is there a GraphQL mutation similar to REST API’s “PUT /repos/:owner/:repo/branches/:branch/protection” to allow force pushes to the protected branch?</p>
<p>With the <a href="https://developer.github.com/v3/repos/branches/#update-branch-protection" rel="nofollow noopener">REST API</a>, I can create a branch protection rule and set allow_force_pushes: true.</p>
<p>How can I do this with GraphQL? I don’t readily see how with <a href="https://developer.github.com/v4/input_object/createbranchprotectionruleinput/" rel="nofollow noopener">CreateBranchProtectionRuleInput </a>or <a href="https://developer.github.com/v4/input_object/updatebranchprotectionruleinput/" rel="nofollow noopener">UpdateBranchProtectionRuleInput</a>.</p>
<p>For added context, in the GitHub UI this is the setting I’m trying to change via GraphQL API:</p>
<p><img src="https://user-images.githubusercontent.com/4142577/181086479-46cff87a-7849-44c6-8f5e-2ed6e7f20842.png" alt="allow-force-pushes.png" data-base62-sha1="dlmSaC0Y9Ls0yCHbEFRGxlCd72f" width="465" height="225"></p>
<p>Thanks</p>

**Comment by douglascayers** [comment]
<p>Per GitHub Support, the GraphQL API doesn’t expose this option.</p>
<blockquote>
<p>Thanks for reaching out and asking us about this! The GraphQL API doesn’t currently expose that field today, though it could change in the future. We don’t have a timeline for if or when that would happen, though we encourage you to subscribe to our changelog: <a href="https://developer.github.com/v4/changelog/" rel="nofollow noopener">https://developer.github.com/v4/changelog/</a></p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>As a workaround, we suggest continuing to use the REST API endpoint for updating that setting.</p>
</blockquote>

**Comment by v1ct0rv** [comment]
<p>For me the problem is REST API does not support wildcards in brach name, so I cannot use it…</p>

**Comment by dmitry-zhuravlev** [comment]
<p>The workaround not really good since it depends on the particular branch. So to fetch “allow force push”/“allow deletions” options you need to query REST API N times where N is the number of the branches in the repository. Also REST API doesn’t return the rule pattern itself and you need to query it in the GraphQL API in additional queries. Completely not efficient and not useful.</p>

**Comment by donatelloOo** [comment, marked as answer]
<p>it seems that latest GitHub as long as GHE 3.0 now supports these properties on GraphQL.</p>

**Reply by douglascayers** [reply]
Thanks, yes I see that now at https://docs.github.com/en/graphql/reference/input-objects#createbranchprotectionruleinput
